src/app/pages/datasets/components/dataset-management/dataset-management.component.ts

```diff
@@ -86,11 +87,22 @@ export class DatasetsManagementComponent implements OnInit {
   }
 
   private createDataSource(datasets: DatasetInTree[]): void {
-    this.dataSource = new IxNestedTreeDataSource<DatasetInTree>(datasets);
-    this.dataSource.filterPredicate = (datasets, query = '') => {
-      return datasets.map((datasetRoot) => {
+    const filteredDatasets = this.initFilterDataSource(datasets);
+    this.dataSource = new IxNestedTreeDataSource<DatasetInTree>(filteredDatasets);
+    this.dataSource.filterPredicate = (filteredDatasets, query = '') => {
+      return filteredDatasets.map((datasetRoot) => {
         return findInTree([datasetRoot], (dataset) => dataset.id.toLowerCase().includes(query.toLowerCase()));
       }).filter(Boolean);
     };
   }
+
+  private initFilterDataSource(datasets: DatasetInTree[]): DatasetInTree[] {
+    const filtered = (datasets: DatasetInTree[]): DatasetInTree[] => {
+      return datasets.filter((dataset) => dataset.type === DatasetType.Filesystem && !dataset.locked).map((dataset) => {
```

Comment:
Does it mean we don't need Zvol datasets (`type === DatasetType.Volume`) in the tree? I'm curious because working on a widget for Zvol datasets.

Reply:
I misunderstood the requirements. 
I have to fix this again.